vendor/github.com/Azure/go-autorest/autorest/adal/token.go

```diff
@@ -687,6 +694,8 @@ func NewServicePrincipalTokenFromAuthorizationCode(oauthConfig OAuthConfig, clie
 }
 
 // NewServicePrincipalTokenFromFederatedToken creates a ServicePrincipalToken from the supplied federated OIDC JWT.
+//
+// Deprecated: Use NewServicePrincipalTokenFromFederatedTokenWithCallback to refresh jwt dynamically.
```

Comment:
Updating the Azure SDK added this deprecation warning which is causing the linter to fail. This is part of the v1 Azure implementation. Since we are replacing this SDK anyway and will delete the code soon, I suggest we ignore it for now.